### PR TITLE
Accept uuids kwargs for weaviate

### LINKED ISSUE #4791: Accept UUID list as an argument to add texts and documents into Weaviate vectorstore
### Feature request

When you call `add_texts` and `add_docuemnts` methods from a Weaviate instance, it always generate UUIDs for you, which is a neat feature

https://github.com/hwchase17/langchain/blob/bee136efa4393219302208a1a458d32129f5d539/langchain/vectorstores/weaviate.py#L137

However, there are specific use cases where you want to generate UUIDs by yourself and pass them via `add_texts` and `add_docuemnts`.
Therefore, I'd like to support `uuids` field in `kwargs` argument to these methods, and use those values instead of generating new ones inside those methods.

### Motivation

Both `add_texts` and `add_documents` methods internally call [batch.add_data_object](https://weaviate-python-client.readthedocs.io/en/stable/weaviate.batch.html#weaviate.batch.Batch.add_data_object) method of a Weaviate client.

The document states as below:

> Add one object to this batch. NOTE: If the UUID of one of the objects already exists then the existing object will be replaced by the new object.

This behavior is extremely useful when you need to update and delete document from a known field of the document.

First of all, Weaviate expects UUIDv3 and UUIDv5 as UUID formats. You can find the information below:
https://weaviate.io/developers/weaviate/more-resources/faq#q-are-there-restrictions-on-uuid-formatting-do-i-have-to-adhere-to-any-standards

And UUIDv5 allows you to generate always the same value based on input string, as if it's a hash algorithm.
https://docs.python.org/2/library/uuid.html

Let's say you have unique identifier of the document, and use it to generate your own UUID.
This way you can directly update, delete or replace documents without searching the documents by metadata.

This will saves your time, your code, and network bandwidth and computer resources.

### Your contribution

I'm attempting to make a PR,

### PR DESCRIPTION
**Pull request by yujiosaka**
# Accept uuids kwargs for weaviate

Fixes #4791

## Before submitting

- [x] added integration tests

didn't add any example notebook as i was afraid the feature was a bit too small to make a dedicated notebook.
i looked up at another Weaviate option called `where_filter`, which had integration tests but not notebook.

so i decided just to add integration tests. feel free to make suggestion.

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

### CODE REVIEW

File: langchain/vectorstores/redis.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def _similarity_search_with_relevance_scores(
         if self.relevance_score_fn is None:
             raise ValueError(
                 "relevance_score_fn must be provided to"
-                " Weaviate constructor to normalize scores"
+                " Redis constructor to normalize scores"
```
Comment (yujiosaka) [inline langchain/vectorstores/redis.py:354]: accidentally found a typo when i was grepping Weaviate in the project.

**Comment by dev2049** [comment]
thanks @yujiosaka!

**Review by JohnsonManuel — COMMENTED** [review]
Issue: The line _id = get_valid_uuid(uuid4()) has a mismatch in expected input and output types. The get_valid_uuid() function expects a str as input, but uuid4() returns a UUID object.

_id = get_valid_uuid(str(uuid4()))

**Comment by yujiosaka** [comment]
@JohnsonManuel 
`get_valid_uuid()` accepts both str and UUID object since weaviate-client v3.3.1
https://weaviate-python-client.readthedocs.io/en/v3.3.1/weaviate.util.html?highlight=get_valid_uuid#weaviate.util.get_valid_uuid

but i can make a PR to support older weaviate-client versions

**Comment by yujiosaka** [comment]
@JohnsonManuel 
tried testing with weaviate-client older versions.
i figured out that langchain uses bunch of methods introduced from v3.17.0 such as `schema.exists` and `tenant` attribute in `Batch.add_data_object`
so i suggest you simply upgrade weaviate client version

**Pull request merged**